src/components/FileUpload.tsx

```diff
@@ -54,16 +55,105 @@ export default function FileUpload() {
         setErrorOpen(false);
     };
 
+    async function getToken() {
+        const token = await firebase.auth().currentUser?.getIdToken();
+
+        if (token === undefined) return {};
+
+        return token;
+    }
+
     useEffect(() => {
         globalThis.uploadedDatasets = datasets;
     }, [datasets]);
 
+    async function uploadDatatset(key: string) {
+        if (key === undefined) {
+            enqueueSnackbar("Something went wrong", {
+                variant: "error"
+            });
+            return;
+        }
+
+        fetch(process.env.NEXT_PUBLIC_GQL_API_UPDATED, {
+            method: "POST",
+            headers: {
+                "content-type": "application/json",
+                Authorization: (await getToken()) as unknown as string
+            },
+            body: JSON.stringify({
+                query: `mutation addDataset($datasetKey: String!){
+                         addDataSet(dataSetKey:$datasetKey){
+                            id
+                            name
+                            datasets
+                        }
+                }
+                `,
+                variables: {
+                    datasetKey: `${process.env.NEXT_PUBLIC_DATASET_API}/${key}`
+                }
+            })
+        })
+            .then((resp) => resp.json())
+            .then((result) => {
+                enqueueSnackbar("Dataset uploaded", {
+                    variant: "success"
+                });
+            })
+            .catch((err) => console.log(err));
+    }
+
     return (
         <>
             <Dropzone
                 onDrop={async (acceptedFiles: File[]) => {
-                    console.log("start");
                     let dsChanged = false;
+                    let fileToUploaded = new FormData();
+                    fileToUploaded.append("upload", acceptedFiles[0]);
+
+                    fetch(process.env.NEXT_PUBLIC_DATASET_API, {
+                        method: "POST",
+                        headers: {
+                            //TODO (verite) I'm sure there is a better way to write this but I'll fix this letter
+                            Authorization:
+                                (await getToken()) as unknown as string
```

Comment:
```suggestion
                                await getToken()
```
If the fix in getToken works then this shouldn't be necessary